pkg/resource/defaultclass_test.go

```diff
@@ -0,0 +1,245 @@
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package resource
+
+import (
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/pkg/errors"
+
+	"k8s.io/apimachinery/pkg/runtime"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+
+	"github.com/crossplaneio/crossplane/pkg/apis/core/v1alpha1"
+	corev1alpha1 "github.com/crossplaneio/crossplane/pkg/apis/core/v1alpha1"
+	"github.com/crossplaneio/crossplane/pkg/meta"
+	"github.com/crossplaneio/crossplane/pkg/test"
+)
+
+var _ reconcile.Reconciler = &DefaultClassReconciler{}
+
+func TestDefaultClassReconcile(t *testing.T) {
+	type args struct {
+		m  manager.Manager
+		of ClaimKind
+	}
+
+	type want struct {
+		result reconcile.Result
+		err    error
+	}
+
+	errBoom := errors.New("boom")
+	errUnexpected := errors.New("unexpected object type")
+
+	cases := map[string]struct {
+		args args
+		want want
+	}{
+		"GetClaimError": {
+			args: args{
+				m: &MockManager{
+					c: &test.MockClient{MockGet: test.NewMockGetFn(errBoom)},
+					s: MockSchemeWith(&MockClaim{}, &MockManaged{}),
+				},
+				of: ClaimKind(MockGVK(&MockClaim{})),
+			},
+			want: want{err: errors.Wrap(errBoom, errGetClaim)},
+		},
+		"ListDefaultClassesError": {
+			args: args{
+				m: &MockManager{
+					c: &test.MockClient{
+						MockGet: test.NewMockGetFn(nil, func(o runtime.Object) error {
+							switch o := o.(type) {
+							case *MockClaim:
+								cm := &MockClaim{}
+								*o = *cm
+								return nil
+							case *MockManaged:
+								return errBoom
+							default:
+								return errUnexpected
+							}
+						}),
+						MockList: test.NewMockListFn(errBoom),
+						MockStatusUpdate: test.NewMockStatusUpdateFn(nil, func(got runtime.Object) error {
+							want := &MockClaim{}
+							want.SetConditions(v1alpha1.ReconcileError(errors.New(errFailedList)))
+							if diff := cmp.Diff(want, got, test.EquateConditions()); diff != "" {
+								t.Errorf("-want, +got:\n%s", diff)
+							}
+							return nil
+						}),
+					},
+					s: MockSchemeWith(&MockClaim{}),
+				},
+				of: ClaimKind(MockGVK(&MockClaim{})),
+			},
+			want: want{result: reconcile.Result{}},
+		},
+		"NoDefaultClass": {
+			args: args{
+				m: &MockManager{
+					c: &test.MockClient{
+						MockGet: test.NewMockGetFn(nil, func(o runtime.Object) error {
+							switch o := o.(type) {
+							case *MockClaim:
+								cm := &MockClaim{}
+								*o = *cm
+								return nil
+							case *MockManaged:
```

Comment:
Do we need this case? I don't think we ever try to get anything that satisfies the `resource.Managed` interface. Ditto below cases.